src/sklearn_compat/_sklearn_compat.py

```diff
@@ -206,8 +206,56 @@ def _is_fitted(estimator, attributes=None, all_or_any=all):
         ]
         return len(fitted_attrs) > 0
 
+    if sklearn_version < parse_version("1.3"):
+        # only define those functions to not raise an error.
+        def process_routing(_obj, _method, /, **kwargs):
+            pass
+
+        def _raise_for_params(params, owner, method):
+            pass
+    else:
+
+        def process_routing(_obj, _method, /, **kwargs):
+            """Validate and route input parameters."""
+            from sklearn.utils._metadata_requests import process_routing
+
+            return process_routing(_obj, _method, other_params=None, **kwargs)
```

Comment:
simply using the new signature by the user solves the issue, not sure why we need to include it here. Using the new signature is backward compatible w/o warnings.

Reply:
OK :). I was lazy to after doing the fix and seen my previous comment in the README. Let me revert and change the code in `imbalanced-learn` :)